sdk/src/client/api/block_builder/input_selection/remainder.rs

```diff
@@ -118,30 +122,77 @@ impl InputSelection {
 
         let native_tokens_diff = get_native_tokens_diff(&input_native_tokens, &output_native_tokens)?;
 
-        if inputs_sum == outputs_sum && native_tokens_diff.is_none() {
+        let mut input_mana = 0;
+
+        for input in &self.selected_inputs {
+            input_mana += input.output.all_mana(
+                &self.protocol_parameters,
+                input.output_id().transaction_id().slot_index(),
+                self.slot_index,
+            )?;
+            // TODO rewards https://github.com/iotaledger/iota-sdk/issues/1310
+        }
+
+        let output_mana = self.outputs.iter().map(|o| o.mana()).sum::<u64>() + self.mana_allotments;
+
+        if input_amount == output_amount && input_mana == output_mana && native_tokens_diff.is_none() {
             log::debug!("No remainder required");
             return Ok((None, storage_deposit_returns));
         }
 
+        let amount_diff = input_amount - output_amount;
+        let mana_diff = input_mana - output_mana;
```

Comment:
Should they use something like checked_sub()?

Reply:
seems like a good idea to me

Reply:
Done